drivers/media/i2c/imx219.c

```diff
@@ -756,48 +796,57 @@ static int imx219_set_pad_format(struct v4l2_subdev *sd,
 	int exposure_max, exposure_def, hblank;
 	struct v4l2_mbus_framefmt *format;
 
-	mode = v4l2_find_nearest_size(supported_modes,
-				      ARRAY_SIZE(supported_modes),
-				      width, height,
-				      fmt->format.width, fmt->format.height);
-
-	imx219_update_pad_format(imx219, mode, &fmt->format, fmt->format.code);
-	format = v4l2_subdev_get_pad_format(sd, sd_state, 0);
-
-	if (imx219->mode == mode && format->code == fmt->format.code)
-		return 0;
-
-	if (fmt->which == V4L2_SUBDEV_FORMAT_ACTIVE) {
-		u32 prev_hts = imx219->mode->width + imx219->hblank->val;
+	if (fmt->pad >= NUM_PADS)
```

Comment:
I don't know if this is an old patch, but why handle the error case early? Why not "if IMAGE, else if METADATA. else ERROR"? At least that way it's obvious where METADATA is handled.

Reply:
Old patch - https://github.com/raspberrypi/linux/commit/e3376fb94fda798d2a322e9c70789286132a1a9f

Upstream don't like two independent pads for image and metadata, and are working on adding support for multiplexed pads (6.7 if we're incredibly lucky, but otherwise 6.8). Little point in polishing old patches that are going to be ditched RSN.